tests/integration/optlyplugins/test_composite_service.go

```diff
@@ -0,0 +1,80 @@
+package optlyplugins
+
+import (
+	"github.com/optimizely/go-sdk/pkg/decision"
+	"github.com/optimizely/go-sdk/pkg/notification"
+	"github.com/optimizely/go-sdk/tests/integration/models"
+)
+
+// TestCompositeService represents a CompositeService with custom implementations
+type TestCompositeService struct {
+	decision.CompositeService
+}
+
+// AddListener - Adds Notification Listeners
+func (c *TestCompositeService) AddListener(params models.APIOptions) (getListenersCalled func() []models.DecisionListener) {
+
+	var listenersCalled []models.DecisionListener
+	getListenersCalled = func() []models.DecisionListener {
+		return listenersCalled
+	}
+	if len(params.Listeners) < 1 {
```

Comment:
Do you need anything else from the `APIOptions`? Can you just pass in the `Listeners`?